internal/daemon/vendor-specific-plugins/common/vspnetutils.go

```diff
@@ -110,3 +132,134 @@ func GetNetDevNameFromPCIeAddr(platform platform.Platform, pcieAddress string) (
 
 	return "", fmt.Errorf("network device not found for PCI address %s", pcieAddress)
 }
+
+// CreateVethPair function to create a veth pair with the given index and InterfaceInfo
+func CreateVethPair(index int) (*VEthPairDeviceInfo, error) {
```

Comment:
this is a general purpose utility (that mainly only depends on netlink). It is good that it is under "common". But it should not be under "vendor-specific-plugins". There should be a utility package that can be used from anywhere (and in this case, the package mainly only depends on netlink).

Reply:
I think the patch should be split in a part that introduces new helper API, and a part that does something about PTL code.

Reply:
We can move the package once we see that the functions we created under vender specific plugin is going to be used elsewhere. Better not assume anything about future use when we can easily move it later.

Reply:
I think the patch is small enough that we don't need to split it up for review.